RFS/src/main/java/org/opensearch/migrations/bulkload/lucene/LuceneDocumentsReader9.java

```diff
@@ -0,0 +1,230 @@
+package org.opensearch.migrations.bulkload.lucene;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.concurrent.Callable;
+
+import org.opensearch.migrations.bulkload.common.RfsLuceneDocument;
+import org.opensearch.migrations.bulkload.common.Uid;
+
+import lombok.Lombok;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.reactivestreams.Publisher;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Schedulers;
+import shadow.lucene9.org.apache.lucene.document.Document;
+import shadow.lucene9.org.apache.lucene.index.DirectoryReader;
+import shadow.lucene9.org.apache.lucene.index.LeafReaderContext;
+import shadow.lucene9.org.apache.lucene.index.SoftDeletesDirectoryReaderWrapper;
+import shadow.lucene9.org.apache.lucene.index.StoredFields;
+import shadow.lucene9.org.apache.lucene.store.FSDirectory;
+import shadow.lucene9.org.apache.lucene.util.BytesRef;
+
+@RequiredArgsConstructor
+@Slf4j
+public class LuceneDocumentsReader9 implements LuceneDocumentsReader {
+
+    protected final Path indexDirectoryPath;
+    protected final boolean softDeletesPossible;
+    protected final String softDeletesField;
+ 
+    /**
+     * There are a variety of states the documents in our Lucene Index can be in; this method extracts those documents
```

Comment:
Could you move this to the class documentation?
The below comments still hold, even if you didn't write them.  If you did - awesome, thanks for the work.  If you didn't, can you 1) make sure that these are only in one place (like an interface that all implementations inherit from)?  You can setup a follow-up task for it too if you like.